.pre-commit-config.yaml

```
@@ -30,7 +32,7 @@ repos:
         args: [--update-only, --title=**Table of Contents**]
   - repo: https://github.com/astral-sh/ruff-pre-commit
     # Ruff for linting and formatting python
-    rev: v0.2.1
+    rev: v0.3.0
```

Because the precommit ci just suggested this change...
```suggestion
    rev: v0.3.2
```